query-engine/connectors/sql-query-connector/src/error.rs

```diff
@@ -181,14 +181,9 @@ pub enum SqlError {
 impl SqlError {
     pub(crate) fn into_connector_error(self, connection_info: &quaint::prelude::ConnectionInfo) -> ConnectorError {
         match self {
-            SqlError::UniqueConstraintViolation { constraint } => ConnectorError {
-                user_facing_error: Some(user_facing_errors::KnownError::new(
-                    user_facing_errors::query_engine::UniqueKeyViolation {
-                        constraint: constraint.clone(),
-                    },
-                )),
-                kind: ErrorKind::UniqueConstraintViolation { constraint },
-            },
+            SqlError::UniqueConstraintViolation { constraint } => {
+                ConnectorError::from_kind(ErrorKind::UniqueConstraintViolation { constraint })
+            }
```

Comment:
We are returning a user facing error. If you pay attention, I haven't added any new kind of errors. I only added a `transient: bool` flag or existing errors so that we keep returning the original errors to users and so that it doesn't make this PR breaking.

If the transient retries don't get rid of the write conflict errors, for instance, then this write conflict error will be forwarded to users. Does that make sense?

Reply:
I got confused but I think we understand "branch" differently, I didn't mean the git branch, but the match branch lines 184 to 186, just for the unique constraint violation error (see the removed lines just above for context)